src/testHelper.ts

```diff
@@ -0,0 +1,164 @@
+// Copyright 2018 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+import { P2pLayer } from "./p2pLayer";
+import {
+    BlockSyncMessage,
+    ResponseMessage,
+    MessageType,
+    IHeadersq,
+    IBodiesq
+} from "./blockSyncMessage";
+import { H256 } from "codechain-sdk/lib/core/H256";
+import { U256 } from "codechain-sdk/lib/core/U256";
+
+export class TestHelper {
+    private p2psocket: P2pLayer;
+
+    constructor(ip: string, port: number) {
+        this.p2psocket = new P2pLayer(ip, port);
+    }
+
+    getP2pSocket(): P2pLayer {
+        return this.p2psocket;
+    }
+
+    async establish() {
+        await this.p2psocket.connect();
+        console.log("Connected\n");
+    }
+
+    // Get block headers from the most recent header response
+    getBlockHeaderResponse(): Array<Array<Buffer>> | null {
+        for (const msg of this.p2psocket
+            .getArrivedExtensionMessage()
+            .reverse()) {
```

Comment:
Why do you reverse it?

Reply:
I found that it's the only way to get the recent message since the p2psocket doesn't remove the arrived messages.
Then I'll change the question.
What happens if there is a new message before it handles the message.

Reply:
Nothing happens. Honestly, there is no need to store past messages